text/0000-deprecate-ember-k.md

```diff
@@ -0,0 +1,102 @@
+- Start Date: 2016-11-18
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+The `Ember.K` utility function is a low level utility that has lost most of its value today.
+
+# Motivation
+
+Let's start explaining what `Ember.K` is.
+
+It is an utility function to avoid boilerplace code and limit the creation of function instances
+in Ember's internals. The source code for this API is the following:
+
+```js
+Ember.K = function() {
+  return this;
+}
+```
+
+In a world of globals, writing `somefn: Ember.K` was effectively shorter
+than writing
+
+```js
+someFn: function() {
+  return this;
+}
+```
+
+and generated fewer function allocations.
+
+However with the introduction of ES6 modules and the modularization of Ember
+in process (#176), keeping this feature would require to design an import path for it.
+
+While doable, the transpiled output is actually bigger then defining the functions
+inline, specially with the ES6 shorthand method syntax, and the perf difference
+of saving a few function allocations is despicable.
+
+The second downside of reusing the same instance in many places is that if for
+some reason the VM deoptimizes that function, that deoptimization is spreaded
+across all the usages of `Ember.K`.
```

Comment:
ya, if this happens. Large chunks of inlined code could be evicted, making many things sad. Although this risk does seem to continue to reduce as JITs get better, if we can avoid it I would recommend we do.